source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -131,6 +131,7 @@ Network::FilterStatus Filter::onData(Network::ListenerFilterBuffer& buffer) {
     const size_t len = raw_slice.len_ - read_;
     read_ = raw_slice.len_;
     ParseState parse_state = parseClientHello(data, len);
+    ENVOY_LOG(trace, "tls inspector: set transport protocol to {}", cb_->socket().detectedTransportProtocol());
```

Comment:
Should we log at where the transport protocol set(before the line 183)?
https://github.com/envoyproxy/envoy/pull/20565/files#diff-3004fa07320c547f4fdd1ba336a346a906d1c4d8b3260216d6625f86dfcf85b5R183

Since there are failure path inside `parseClientHello`

Reply:
It is only the SSL case I think.

Reply:
For example, if the code is going to line 172, then you will get a empty value for `detectedTransportProtocol`. I'm not sure this is what people want.